django_pin_auth/auth_backend.py

```diff
@@ -0,0 +1,31 @@
+from django.contrib.auth import get_user_model
+
+from .models import SingleUseToken
+
+class PinBackend(object):
+    """Authentication backend based on pin value."""
+    def authenticate(self, request, email=None, pin=None):
```

Comment:
What do you think if instead of checking for User and Token in 2 steps

We just grab the token:

```python3
try:
    token = SingleUseToken.objects.select_related('user').get(user__email=email, token=pin)
except SingleUseToken.DoesNotExist:
    return None
```

Reply:
Using the USERNAME_FIELD also

Reply:
See a4a5ca393ffc553202b266bdea790768a119f8f8